src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -407,358 +407,6 @@ def get_arguments_loader(context, getter_op, cmd_args=None, operation_group=None
     return cmd_args
 
 
-# pylint: disable=too-many-statements
-def _cli_generic_update_command(context, name, getter_op, setter_op, setter_arg_name='parameters',
-                                child_collection_prop_name=None, child_collection_key='name',
-                                child_arg_name='item_name', custom_function_op=None, **kwargs):
```

Comment:
I agree and like this refactoring. The original code is obviously abusing closure and makes the code difficult to read.

Reply:
But still, even in closure, the values can be retrieved using [`inspect.getclosurevars(func)`](https://docs.python.org/3/library/inspect.html#inspect.getclosurevars).